internal/action/workspace.go

```diff
@@ -143,15 +144,68 @@ func NewWorkspaceResource(ctx context.Context, client *tfe.Client, workspaces []
 	ws.SSHKeyID = config.SSHKeyID
 	ws.WorkingDirectory = config.WorkingDirectory
 
+	if err := SetTags(ws, config.Tags); err != nil {
+		return nil, err
+	}
+
 	return ws, nil
 }
 
+type Tag string
+type Tags []Tag
+
 type TeamDataResource struct {
 	ForEach      map[string]TeamDataResource `json:"for_each,omitempty"`
 	Name         string                      `json:"name"`
 	Organization string                      `json:"organization"`
 }
 
+// SetTags adds workspace tags to the passed module
+func SetTags(module *tfeprovider.Workspace, tags map[string]Tags) error {
+	if len(tags) == 0 {
+		return nil
+	}
+
+	// Marshals the tags into a JSON object which we will use as HCL in a for_each lookup.
+	// This is a result of using for_each for the workspace resource itself, which could be avoided
+	// if we were to use distinct resources for each workspace as each could get their own tag set.
+	// This implementation is an alternative approach to putting the tags in a local variable, as that splits
+	// the tag logic up quite a bit, i.e. we would be setting the tag reference here on the workspace,
+	// and the tags lookup map elsewhere on the module.
```

Comment:
@bendrucker after poking a bit, I added a comment which explains some of the awkwardness with adding a local lookup to the module. I'm not definitively opposed to the approach, just couldn't reason out a very elegant way to do it. 

- Passing the module itself  into the NewWorkspaceResource call for the purpose of adding a local variable seems backwards, as the module is supposed to own the workspace resource. This seemed like the best way to me but.. still not great imo
- Creating the local var on the module and passing in a reference to the tags instead of the tags themselves into NewWorkspaceResource leeks because it implies knowledge of how the workspace resource will handle the tags input
- Creating the local var on the module and then letting NewWorkspaceResource reference the tags via a magically shared var name seems brittle to me

all seemed a bit messy, this approach at least keeps the implementation all in one spot without passing unneeded references around. lmk what you think!